pkg/providers/v1/aws.go

```diff
@@ -1223,19 +1223,16 @@ func init() {
 			}
 		} else {
 			klog.Infof("Using AWS assumed role %v", cfg.Global.RoleARN)
-			provider = &stscreds.AssumeRoleProvider{
-				Client:  sts.New(sess),
-				RoleARN: cfg.Global.RoleARN,
-			}
+			creds := credentials.NewChainCredentials(
+				[]credentials.Provider{
+					&credentials.EnvProvider{},
+					assumeRoleProvider(&stscreds.AssumeRoleProvider{
+						Client:  sts.New(sess),
+						RoleARN: cfg.Global.RoleARN,
+					}),
+				})
 		}
 
-		creds := credentials.NewChainCredentials(
```

Comment:
if cfg.Global.RoleARN == "", then creds is never set.

Try only changing this: 

```
provider = assumeRoleProvider(&stscreds.AssumeRoleProvider{
				Client:  sts.New(sess),
				RoleARN: cfg.Global.RoleARN,
			})
``